packages/react/src/ReactElementValidator.js

```diff
@@ -226,9 +236,44 @@ function validatePropTypes(element) {
   }
 }
 
+/**
+ * Given a fragment, validate that it can only be provided with fragment props
+ * @param {ReactElement} fragment
+ */
+function validateFragmentProps(fragment) {
+  currentlyValidatingElement = fragment;
+
+  for (const key of Object.keys(fragment.props)) {
```

Comment:
We shouldn't have used this. It introduces reliance on `Symbol` which we don't currently require.

I think we should add a lint against `for ... of` as it's easy to use accidentally and it's not obvious what it compiles down to.